src/preprocess/links.rs

```diff
@@ -46,50 +70,73 @@ impl<'a> Link<'a> {
                 let props: Vec<&str> = path_props.collect();
 
                 match (typ.as_str(), file_path) {
-                    ("include", Some(pth)) => Some(LinkType::Include(pth)),
+                    ("include", Some(pth)) => Self::parse_include_path(pth),
                     ("playpen", Some(pth)) => Some(LinkType::Playpen(pth, props)),
                     _ => None,
                 }
             }
-            (Some(mat), None, None) if mat.as_str().starts_with(ESCAPE_CHAR) => {
-                Some(LinkType::Escaped)
-            }
+            (Some(mat), None, None) if mat.as_str().starts_with(ESCAPE_CHAR) => Some(
+                LinkType::Escaped,
+            ),
             _ => None,
         };
 
         link_type.and_then(|lnk| {
             cap.get(0).map(|mat| {
-                               Link {
-                                   start_index: mat.start(),
-                                   end_index: mat.end(),
-                                   link: lnk,
-                                   link_text: mat.as_str(),
-                               }
-                           })
+                Link {
+                    start_index: mat.start(),
+                    end_index: mat.end(),
+                    link: lnk,
+                    link_text: mat.as_str(),
+                }
+            })
         })
     }
 
+    fn join_lines<'b, I: IntoIterator<Item = &'b str>>(lines: I) -> String {
```

Comment:
Most likely, better to use itertools::intersperse, however, I am not sure if it is ok to add another dependency to the crate.

Reply:
Added itertools.